scrapper.py

```diff
@@ -25,39 +36,175 @@ class Crawler:
+                    continue
+            if not pdf_link:
+                print('Error')
+            response_pdf = requests.get(pdf_link, HEADERS)
+            # print(response_pdf)
+            # print("ЗДЕСЬ КОНЕЦ СТАТЬИИИИИИИИИИИИИИИИИИИИИ")
+            pdf_bs = BeautifulSoup(response_pdf.text, 'lxml')
+
+            download_pdf = pdf_bs.find('a', {'class': 'download'})['href']
+
+            pdf = PDFRawFile(download_pdf, self.article_id)
+
+            pdf.download()
+            self.article.text = pdf.get_text()
+        except requests.exceptions.MissingSchema:
+            print(article_bs)
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Extract Meta information
+        """
+        try:
+            article_author_bs = article_bs.find('span', {'class': 'name'})
+        except AssertionError:
+            article_authors_bs = article_bs.find('ul', {'class': 'item authors'})
+            article_authors_list_bs = article_authors_bs.findall('span', {'class': 'name'})
+            article_author_bs = ' '.join(article_authors_list_bs)
+        if not article_author_bs:
+            article_author_bs = 'Not Found'
+        self.article.author = article_author_bs
+
+        article_title_bs = article_bs.find('h1', {'class': 'page_title'})
+        self.article.title = article_title_bs
+
+        date_holder_bs = article_bs.find('div', {'class': 'item published'})
+        article_date_bs = date_holder_bs.find('div', {'class': 'value'})
+        self.article.date = article_date_bs
+
+    def parse(self):
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = pathlib.Path(base_path)
+    if path.exists():
+        shutil.rmtree(base_path)
+    path.mkdir(exist_ok=True, parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    if "total_articles_to_find_and_parse" not in config:
+        raise IncorrectNumberOfArticlesError
+
+    if "seed_urls" not in config:
+        raise IncorrectURLError
+
+    seed_urls = config["seed_urls"]
+
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    for seed_url in seed_urls:
+        if not re.match(r'https?://', seed_url):
+            raise IncorrectURLError
+
+    total_articles = config["total_articles_to_find_and_parse"]
+
+    if not isinstance(total_articles, int) or total_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if total_articles > 300:
+        raise NumberOfArticlesOutOfRangeError
+
+    prepare_environment(ASSETS_PATH)
```

Comment:
follow the single responsibility principle - a function that *validates* cannot *prepare environment*. therefore, give it a thought on where this function should be properly called

Reply:
done